src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -325,31 +319,7 @@ private void TagHelperDirective(string keyword, Func<string, ISpanChunkGenerator
                 // etc.
                 AcceptUntil(CSharpSymbolType.NewLine);
 
-                // Pull out the value minus the spaces at the end
-                var rawValue = Span.GetContent().Value.TrimEnd();
-                var startsWithQuote = rawValue.StartsWith("\"", StringComparison.OrdinalIgnoreCase);
-
-                // If the value starts with a quote then we should generate appropriate C# code to colorize the value.
-                if (startsWithQuote)
-                {
-                    // Set up chunk generation
-                    // The generated chunk of this chunk generator is picked up by CSharpDesignTimeHelpersVisitor which
-                    // renders the C# to colorize the user provided value. We trim the quotes around the user's value
-                    // so when we render the code we can project the users value into double quotes to not invoke C#
-                    // IntelliSense.
-                    Span.ChunkGenerator = buildChunkGenerator(rawValue.Trim('"'));
-                }
-
-                // We expect the directive to be surrounded in quotes.
-                // The format for tag helper directives are: @directivename "SomeValue"
-                if (!startsWithQuote ||
-                    !rawValue.EndsWith("\"", StringComparison.OrdinalIgnoreCase))
-                {
-                    Context.OnError(
-                        startLocation,
-                        RazorResources.FormatParseError_DirectiveMustBeSurroundedByQuotes(keyword),
-                        rawValue.Length);
-                }
+                Span.ChunkGenerator = chunkGenerator;
```

Comment:
Any reason not to update the `ChunkGenerator` in the error case as well? I.e. could we move this below the `if`?